lib/charms/vault_k8s/v0/vault_client.py

```diff
@@ -416,3 +447,100 @@ def get_num_raft_peers(self) -> int:
         """Return the number of raft peers."""
         raft_config = self._client.sys.read_raft_config()
         return len(raft_config["data"]["config"]["servers"])
+
+    def _get_autounseal_policy_name(self, relation_id: int) -> str:
+        """Return the policy name for the given relation id."""
+        return f"charm-autounseal-{relation_id}"
+
+    def _get_autounseal_approle_name(self, relation_id: int) -> str:
+        """Return the approle name for the given relation id."""
+        return f"charm-autounseal-{relation_id}"
```

Comment:
Do those helpers belong in this lib?

Reply:
I guess the alternative would be to make the charm pass in the name of each of them (key name, policy name, approle name)? I felt like this kept the interface a little cleaner, and since it is an internal lib, we're not beholden to it.

Do you think it makes more sense to pass each one of these in as a parameter?

Reply:
I think this belongs in the autounseal library. KV also has its own format for approles, and it doesn't make sense to have both of them here when it's just KV and autounseal that will be using these functions.

Reply:
@kayra1 as mentioned in [another comment](https://github.com/canonical/vault-k8s-operator/pull/365#discussion_r1625205980), I don't think we can do this without creating cross-library dependencies. Although this might be technically feasible (I'm not sure), I haven't seen it done with any other charm libraries and there's probably a bit of discussion that needs to be had before we start investigating that. This is why, for example, some KV specific stuff already exists in `vault_client` -- because it needs to interact with the Vault cluster directly.

Reply:
I'm going to leave this as-is for now. I don't want to push the charm to having too much knowledge about the internal workings of Vault if it doesn't need to be there until we've had some discussion about https://warthogs.atlassian.net/browse/TLSENG-269